sourcetracker/_sourcetracker.py

```diff
@@ -618,296 +626,105 @@ def gibbs_sampler(cp, sink, restarts, draws_per_restart, burnin, delay):
 
 
-def sinks_and_sources(sample_metadata, column_header="SourceSink",
-                      source_value='source', sink_value='sink'):
-    '''Return lists of source and sink samples.
-
-    Notes
-    -----
-    This function assumes that source and sink samples will be delineated by
-    having the value `source` or `sink` under the header `SourceSink` in
-    `sample_metadata`.
-
-    Parameters
-    ----------
-    sample_metadata : dict
-        Dictionary containing sample metadata in QIIME 1 sample metadata
-        mapping file format.
-    column_header : str, optional
-        Column in the mapping file that describes where a sample is a source
-        or a sink.
-    source_value : str, optional
-        Value that indicates a sample is a source.
-    sink_value : str, optional
-        Value that indicates a sample is a sink.
-
-    Returns
-    -------
-    source_samples : list
-        Samples that will be collapsed and used as 'Sources'.
-    sink_samples : list
-        Samples that will be used as 'Sinks'.
-    '''
-    sink_samples = []
-    source_samples = []
-    for sample, md in sample_metadata.items():
-        if md[column_header] == sink_value:
-            sink_samples.append(sample)
-        elif md[column_header] == source_value:
-            source_samples.append(sample)
-        else:
-            pass
-    return source_samples, sink_samples
-
-
-def _cli_sync_biom_and_sample_metadata(sample_metadata, biom_table):
-    """Reduce mapping file dict and biom table to shared samples.
-
-    Notes
-    -----
-    This function is used by the command line interface (CLI) to allow mapping
-    files and biom tables which have disjoint sets of samples to be used in the
-    script without forcing the user to keep multiple copies of either.
-
-    Parameters
-    ----------
-    sample_metadata : dict
-        Traditional parsed QIIME mapping file, i.e. nested dictionary with
-        sample ids as keys to a dictionary with mapping file headers as keys
-        and value for that sample the entries under those headers for that
-        sample.
-    biom_table : biom.Table.table
-        Biom table.
-
-    Returns
-    -------
-    nbiom_table : biom.Table.table
-        Biom table with shared samples only.
-    nsample_metadata : dict
-        Sample metadata dictionary with only shared samples.
-
-    Raises
-    ------
-    ValueError
-            If there are no shared samples a ValueError is raised.
-    """
-    mf_samples = set(sample_metadata)
-    biom_table_samples = set(biom_table.ids(axis='sample'))
-    if mf_samples == biom_table_samples:
-        return sample_metadata, biom_table
+def _gibbs_loo(sources, alpha1, alpha2, beta, restarts, draws_per_restart,
+               burnin, delay, cluster=None):
+    '''Gibb's LOO sampling API - see _gibbs for documentation.'''
+    def f(cp_and_sink):
+        from sourcetracker._sourcetracker import gibbs_sampler
```

Comment:
I'd put a comment in about why this import is here. (You or someone else is going to be wondering in the future.)